test/http-context.test.js

```diff
@@ -48,28 +63,79 @@ describe('HttpContext', function() {
       }));
     });
 
-    describe('arguments without a defined type (or any)', function() {
-      it('should coerce boolean strings into actual booleans', givenMethodExpectArg({
+    describe('don\'t coerce arguments without a defined type (or any) in JSON', function() {
+      it('should not coerce boolean strings into actual booleans', givenMethodExpectArg({
```

Comment:
This breaks backwards compat. I understand why this change makes sense, but it also makes things like this impossible in loopback:

```txt
GET /todos?filter[where][done]=true
```

Obviously we need a bit more sophisticated type declaration support... but when you don't even know the structure (eg. a nested property of an object that) it gets a bit tricky. I'm open to suggestions to overhaul this type system if it means better support for dynamic queries in URLs. Maybe there is some syntax out there that caries more type information?